.github/workflows/hardware-examples.yml

```diff
@@ -0,0 +1,39 @@
+name: hardware-examples
```

Comment:
The nightly compiler seems to be required to use an allocator in no_std programs, so I split this into a separate CI script.